packages/gatsby-source-contentful/src/normalize.js

```diff
@@ -764,7 +763,15 @@ export const createAssetNodes = ({
     // The content of an entry is guaranteed to be updated if and only if the .sys.updatedAt field changed
     assetNode.internal.contentDigest = assetItem.sys.updatedAt
 
-    createNodePromises.push(createNode(assetNode))
+    createNodePromises.push(
+      createNode(assetNode).then(() => {
+        return {
+          contentful_id: assetItem.sys.id,
+          node_locale: locale.code,
+          file: assetNode.file,
+        }
+      })
+    )
```

Comment:
I think we need to retain full node to later call `createNodeField` with it as argument. We would get whole nodes with `getNodesByType` anyway, so I think we do need to skip this memory optim for now. With code as is we get following error (in particular because our "node" doesn't have `internal`). 

```
error "gatsby-source-contentful" threw an error while running the sourceNodes lifecycle:

Cannot read property 'fieldOwners' of undefined


  TypeError: Cannot read property 'fieldOwners' of undefined
  
  - public.js:941 actions.createNodeField
    [contentful]/[gatsby]/src/redux/actions/public.js:941:22
  
  - redux.js:554 boundActionCreator
    [contentful]/[redux]/lib/redux.js:554:35
  
  - api-runner-node.js:79 doubleBoundActionCreators.<computed>
    [contentful]/[gatsby]/src/utils/api-runner-node.js:79:20
  
  - download-contentful-assets.js:106 
    [contentful]/[gatsby-source-contentful]/download-contentful-assets.js:106:7
  
  - download-contentful-assets.js:22 task
    [contentful]/[gatsby-source-contentful]/download-contentful-assets.js:22:7
  
  - async Promise.all
  
  - download-contentful-assets.js:26 distributeWorkload
    [contentful]/[gatsby-source-contentful]/download-contentful-assets.js:26:3
  
  - download-contentful-assets.js:55 downloadContentfulAssets
    [contentful]/[gatsby-source-contentful]/download-contentful-assets.js:55:3
  
  - source-nodes.js:379 Object.sourceNodes
    [contentful]/[gatsby-source-contentful]/source-nodes.js:379:5
  
  - api-runner-node.js:462 runAPI
    [contentful]/[gatsby]/src/utils/api-runner-node.js:462:16
  

not finished source and transform nodes - 2.803s
```